### ISSUE TITLE
Problems compiling gepetto-viewer-corba on python3

### ISSUE THREAD

**Issue by jviereck** [Gepetto/gepetto-viewer-corba#121]
Hi folks,

when I try to compile the source from master for python3, I get

```
$ cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/opt/openrobots_py35
-- omniORB4 >= 4.1.4 is required.
-- Checking for module 'omniORB4 >= 4.1.4'
--   Found omniORB4 , version 4.2.3
-- Pkg-config module omniORB4 v4.2.3 has been detected with success.
-- PythonInterp: /usr/bin/python
-- PythonLibraries: /usr/lib/x86_64-linux-gnu/libpython2.7.so
-- PythonLibraryDirs: /usr/lib/x86_64-linux-gnu
-- PythonLibVersionString: 2.7.12
-- gepetto-viewer > 4.3.0 is required.
-- Checking for module 'gepetto-viewer > 4.3.0'
--   Found gepetto-viewer > 4.3.0, version 4.4.0
-- Pkg-config module gepetto-viewer v4.4.0 has been detected with success.
-- Looking for Qt 4.
-- Could NOT find PythonQt (missing:  PYTHONQT_LIBRARY PYTHONQT_INCLUDE_DIR QtAll) 
-- qgv is optional.
-- Checking for module 'qgv'
--   No package 'qgv' found
CMake Error at CMakeLists.txt:118 (INCLUDE):
  include called with wrong number of arguments.  include() only takes one
  file.


CMake Error at plugins/CMakeLists.txt:20 (INCLUDE):
  include called with wrong number of arguments.  include() only takes one
  file.


CMake Error at plugins/omniorbserver/CMakeLists.txt:25 (GEPETTO_GUI_PLUGIN):
  Unknown CMake command "GEPETTO_GUI_PLUGIN".


-- Configuring incomplete, errors occurred!
See also "/home/jviereck/dev/tools/gepetto-viewer-corba/build/CMakeFiles/CMakeOutput.log".
See also "/home/jviereck/dev/tools/gepetto-viewer-corba/build/CMakeFiles/CMakeError.log".
```

It looks like the variable `GEPETTO_VIEWER_CMAKE_PLUGIN` is not set, which leads to the empty include at `CMakeLists.txt:118`. Also, there is the missing `GEPETTO_GUI_PLUGIN` command.

Any idea how to fix this?

Best,
Julian

**Comment by nim65s** [comment]
Hi @jviereck 

First of all, as usual, we probably did a bad job at communicating about that, but I guess that you don't need this package anymore, because the `gepetto-gui` binary is now in gepetto-viewer.

Also, with the release of Pinocchio v2.1.2 on robotpkg, we updated gepetto-viewer to 4.5.0 and gepetto-viewer-corba to 5.2.0 on robotpkg binary repository, so probably you only need to run:
`sudo apt install -qqy robotpkg-py35-qt4-gepetto-viewer`.


Then, about your issue, `GEPETTO_GUI_PLUGIN` is defined in gepetto-viewer >= v4.5.0.

**Comment by jmirabel** [comment]
I think the version of GV is too old in your case. Moreover, the detected Python version is 2.7 and not 3.5...

@nim65s You still need gepetto-viewer-corba if you use the CORBA interface.

To compile GV-corba with Python3, you need the Omniorb Python backend to be in Python 3. @nim65s is it available now on robotpkg ?

**Comment by nim65s** [comment]
All Python 2/3 - Qt 4/5 are available (except Qt 5 on 14.04), but I did not test them yet

**Comment by nim65s** [comment]
Cf http://robotpkg.openrobots.org/rbulk/robotpkg/graphics/gepetto-viewer/ & http://robotpkg.openrobots.org/rbulk/robotpkg/graphics/gepetto-viewer-corba/

**Comment by jmirabel** [comment]
How do you install Omniorb with Python 3 support ?

**Comment by nim65s** [comment]
I'm not sure to understand your question.

If you are asking how to install omniORBpy st. it can generate Python 3 bindings, well, it's built in, no matter if you install omniORBpy with Python 2 or Python 3 it should be able to generate Python 2 and Python 3 bindings.

If you are asking how did I manage to make gepetto-viewer-corba compile with Python 3… I added a few tweaks, and will open PR here soon.

**Comment by jmirabel** [comment]
I mean the same file as the one listed by `dpkg -l python-omniorb` but for Python 3.

**Comment by nim65s** [comment]
I am using those files. I don't install other ones.

**Comment by jmirabel** [comment]
But they are in `/usr/lib/python2.7`...

**Comment by nim65s** [comment]
Yes they are. And omniidl is run in python2, and correctly finds them in this place. But it is still able to generate python 3 code.

**Comment by jmirabel** [comment]
Generation may work but execution won't.

**Comment by nim65s** [comment]
I see, thanks for the notification :)

**Comment by jmirabel** [comment]
@jviereck If you want GV with Python 3, you must download and compile omniorb and omniorbpy yourself. It is straightforward but not automatic...

**Comment by jviereck** [comment]
@nim65s thanks for your pointer to `sudo apt install -qqy robotpkg-py35-qt4-gepetto-viewer`. I was able to install this version 4.5.0 from `apt-get`.

Two things:
- Is it possible to run the python27 and python35 of gepetto viewer at the same time?
- The package `robotpkg-py35-qt4-gepetto-viewer` doesn't contain any python files (`dpkg-query -L robotpkg-py35-qt4-gepetto-viewer`). Is this expected?

Best,
Julian

**Comment by jmirabel** [comment]
> @nim65s thanks for your pointer to `sudo apt install -qqy robotpkg-py35-qt4-gepetto-viewer`. I was able to install this version 4.5.0 from `apt-get`.

`robotpkg-py35-qt4-gepetto-viewer` will work but you still need `robotpkg-py35-qt4-gepetto-viewer-corba` to use the Python API. And the latter one is available but there is a missing dependency. So installation should work but execution should fail...

> 
> Two things:
> 
>     * Is it possible to run the python27 and python35 of gepetto viewer at the same time?
> 

I  think they won't right now. There is some *tiny* changes to be applied first. I let @nim65s confirm.

>     * The package `robotpkg-py35-qt4-gepetto-viewer` doesn't contain any python files (`dpkg-query -L robotpkg-py35-qt4-gepetto-viewer`). Is this expected?

Yes, it is. There will be some in the future. Python version, for this package, only concerns the internal Python console (to create Qt widgets from Python plugins).

**Comment by jviereck** [comment]
> robotpkg-py35-qt4-gepetto-viewer will work but you still need robotpkg-py35-qt4-gepetto-viewer-corba to use the Python API. And the latter one is available but there is a missing dependency. So installation should work but execution should fail...

Thanks for the tip about `robotpkg-py35-qt4-gepetto-viewer-corba` @jmirabel !I've installed the package, copy-pasted the installed python3.5/gepetto files into my openrobots_py35 folder and tried to execute it. It seems there is an `graphical_interface_idl` module missing at the top level of 

```
ImportError                               Traceback (most recent call last)
~/Downloads/basic_code/display_quadruped.py in <module>
     12 
     13 # Setup the display (connection to gepetto viewer) and load the robot model.
---> 14 robot.initDisplay(loadModel=True)
     15 
     16 # Create a first initial position for the robot. Both legs are bent inwards.

/opt/openrobots_py35/lib/python3.5/site-packages/pinocchio/robot_wrapper.py in initDisplay(self, windowName, sceneName, loadModel)
```
    279         # Set viewer to use to gepetto-gui.
    280         self.used_viewer = 'gepetto-gui'
--> 281         import gepetto.corbaserver
    282         try:
    283             self.viewer = gepetto.corbaserver.Client()

/opt/openrobots_py35/lib/python3.5/site-packages/gepetto/__init__.py in <module>
      7 
      8 # ** 1. Stub files contributing to this module
----> 9 import graphical_interface_idl
     10 
     11 # ** 2. Sub-modules

ImportError: No module named 'graphical_interface_idl'
```

I've uncommented the `import graphical_interface_idl` and fixed a few python3 imports (e.g. replace `import corbaserver` with `from . import corbaserver`). However, then I still end up with this error:

```
/opt/openrobots_py35/lib/python3.5/site-packages/gepetto/corbaserver/graphical_interface_idl.py in <module>
      5 _0_CORBA = CORBA
      6 
----> 7 _omnipy.checkVersion(3,0, __file__)
      8 
      9 

ImportError: Stubs not compatible with omniORBpy version 4.2.
```

Does the omniORBpy in version 4.2 work for you with the current python3 packages? Note that I have installed omniORBpy 4.2.3 for python3.

**Comment by jmirabel** [comment]
First of all, what you do is a hack. Binaries for gepetto-viewer-corba for Python 3 are not functional yet. I believe we are almost there. If you are not wiling to wait their availability, then below are some hints on how to proceed.

I tried once to install the latest omniORB, with Python 3 support. I compiled gepetto-viewer-corba with Python 3. I managed to make it work without much effort. IMHO, compilation should be your preferred approach.

The error you have seems to be that the Python back-end was generated with omniORB 4.1, and not 4.2. Uninstall robotpkg-omniorb if installed.
